### ISSUE TITLE
Compiler Regression: Obsolete shadowing collection initializers don't warn

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#3525]
Source:

```C#
class C : List<int> {
	[Obsolete]
	new void Add(int x) => new C { 2 };
}
```

http://tryroslyn.azurewebsites.net/#b:master/f:sr/K4Zwlgdg5gBAygTxAFwKYFsDcAoUlaIoYB0AwgPYA2lqAxsmORCMQOKoSoBOYtOtlAIYgQMUjABcMADJgUAHkjIAfDADe2AJABtAPIAjEFVRoAuls4B3GADdyYACYwAgg4cAKJTAAeAShgAvKpWYuowAEwwAL44UUAA=

Under Roslyn, this incorrectly compiles with no warnings.

The legacy compiler correctly warns CS1064.

**Comment by gafter** [comment]
The compiler is not supposed to give a diagnostic when using an obsolete member from the body of an obsolete member. You found a bug in the old compiler, not Roslyn. As you can see, it is fixed in Roslyn.
